### ISSUE TITLE
Move "how to" guides to new repo

### ISSUE THREAD

**Issue by adaaaam** [package-url/purl-spec#334]
To better clarify the specification, I suggest we move the `How to build purl string from its components` and `How to parse a purl string in its components` sections into a new file. These are important to support the PURL specification but not part of the specification and better organized as appendices to the specification.

**Comment by adaaaam** [comment]
Another step in this process is reviewing any rules in the "how to" guides and confirming that are included in the `Rules for each purl component` section.

**Comment by matt-phylum** [comment]
I think with the current state of the specification they are an integral part of the specification and removing them will lead to ambiguity, particularly with weird PURLs like `pkg:npm/@angular/cli` or `pkg:generic/name@v/1`. Maybe adding more tests to the test suite would be better than trying to make everyone manually review their implementations against the steps in the file (many implementations, including official implementations, do not do follow this procedure).

Other, better known specifications follow this same pattern of including a pseudocode implementation of the spec in the spec to ensure that implementations behave consistently for edge cases.

**Comment by adaaaam** [comment]
@matt-phylum agreed that they are important to include for clarifying the processes of building and parsing PURLs. this issue is suggesting moving them to the end of the specification to improve the readability of the spec itself.

we are planning work to add more tests to the test suite along with a PURL validation mechanism.

**Comment by mjherzog** [comment]
We have decided to handle the topic of building and parsing PURLs and PURL Type details from the Ecma Standard to the Core Spec. This means that they will be documented separately from the current https://github.com/package-url/purl-spec/blob/main/PURL-SPECIFICATION.rst document when we reset that file to PURL-STANDARD to include only the definition of PURL components, the character-encoding rules and permitted characters.
